apps/banners/context_processors.py

```diff
@@ -0,0 +1,17 @@
+from .services import BannerService
+
+
+def banner_context(request):
+    """
+    Context processor that makes banners available in all templates.
+    Uses the banner_location set by the middleware and excludes dismissed banners.
+    """
+    location = getattr(request, "banner_location", None)
+    context = {"banners": []}
+    if location:
+        location_banners = BannerService.get_banner_context(request=request, location=location)
+        context["banners"].extend(location_banners.get("banners", []))
+    global_banners = BannerService.get_banner_context(request=request, location="global")
```

Comment:
rather than doing a separate query you could make the location filter and OR filter:

`.filter(Q(location=location) | Q(location="global"))`